pkg/netceptor/packetconn.go

```diff
@@ -7,9 +7,25 @@ import (
 	"reflect"
 	"time"
 
+	"github.com/ansible/receptor/pkg/logger"
 	"github.com/ansible/receptor/pkg/utils"
 )
 
+type PacketConner interface {
+	SetHopsToLive(hopsToLive byte)
+	SubscribeUnreachable(doneChan chan struct{}) chan UnreachableNotification
+	ReadFrom(p []byte) (int, net.Addr, error)
+	WriteTo(p []byte, addr net.Addr) (n int, err error)
+	LocalAddr() net.Addr
+	Close() error
+	SetDeadline(t time.Time) error
+	SetReadDeadline(t time.Time) error
+	SetWriteDeadline(t time.Time) error
+	GetCancel() *context.CancelFunc
+	GetLocalService() string
+	GetNetceptorLogger() *logger.ReceptorLogger
```

Comment:
The logger should be one that is re referenced throughout the application, i suggest changing this name to `GetLogger()` or just `Logger()` as it has the same meaning